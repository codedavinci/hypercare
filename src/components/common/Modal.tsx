import { ReactNode } from "react";

type ModalProps = {
  isOpen: boolean;
  onClose: () => void;
  children: ReactNode;
};

function Modal({ isOpen, onClose, children }: ModalProps) {
  if (!isOpen) return null;

  return (
    <div
      className="fixed inset-0 z-50 flex items-center justify-center bg-gray-900 bg-opacity-75"
      id="modal"
    >
      <div className="relative w-3/4  max-w-4xl  bg-white p-4 rounded-3xl overflow-y-auto">
        <button
          className="btn btn-sm btn-circle btn-ghost absolute right-2 top-2"
          onClick={onClose}
          role="close"
          id="close"
        >
          <svg
            xmlns="http://www.w3.org/2000/svg"
            className="h-6 w-6"
            fill="none"
            viewBox="0 0 24 24"
            stroke="currentColor"
          >
            <path
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth={2}
              d="M6 18L18 6M6 6l12 12"
            />
          </svg>
        </button>
        {children}
      </div>
    </div>
  );
}

Modal.displayName = "Modal";

export default Modal;
