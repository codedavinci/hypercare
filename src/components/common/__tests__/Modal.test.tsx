import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import Modal from "../Modal";

describe("Modal Component", () => {
  const sampleChildren = <div>Modal Content</div>;

  it("should render correctly when isOpen is true", () => {
    const { asFragment } = render(
      <Modal isOpen={true} onClose={() => {}}>
        {sampleChildren}
      </Modal>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText("Modal Content")).toBeInTheDocument();
  });

  it("should not render when isOpen is false", () => {
    const { asFragment } = render(
      <Modal isOpen={false} onClose={() => {}}>
        {sampleChildren}
      </Modal>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.queryByText("Modal Content")).not.toBeInTheDocument();
  });

  it("should call onClose when close button is clicked", () => {
    const onClose = jest.fn();
    render(
      <Modal isOpen={true} onClose={onClose}>
        {sampleChildren}
      </Modal>
    );
    fireEvent.click(screen.getByRole("close"));
    expect(onClose).toHaveBeenCalledTimes(1);
  });
});
