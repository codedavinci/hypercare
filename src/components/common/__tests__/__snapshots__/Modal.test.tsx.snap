// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should not render when isOpen is false 1`] = `<DocumentFragment />`;

exports[`Modal Component should render correctly when isOpen is true 1`] = `
<DocumentFragment>
  <div
    class="fixed inset-0 z-50 flex items-center justify-center bg-gray-900 bg-opacity-75"
    id="modal"
  >
    <div
      class="relative w-3/4  max-w-4xl  bg-white p-4 rounded-3xl overflow-y-auto"
    >
      <button
        class="btn btn-sm btn-circle btn-ghost absolute right-2 top-2"
        id="close"
        role="close"
      >
        <svg
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18L18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
      <div>
        Modal Content
      </div>
    </div>
  </div>
</DocumentFragment>
`;
